.github/workflows/test.yml

```diff
@@ -23,17 +43,27 @@ jobs:
           - "5.22"
           - "5.20"
 
-    container:
-      image: perldocker/perl-tester:${{ matrix.perl-version }}
-
+    name: Perl ${{ matrix.perl }} on ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v2
 
+      - name: Set up perl
+        uses: shogo82148/actions-setup-perl@v1
+        with:
+          perl-version: ${{ matrix.perl }}
+
+      - uses: actions/download-artifact@master
+        with:
+          name: build_dir
+          path: .
+
+      - run: cpm install -g
+
       - name: Install latest perltidy
         run: cpm install -g Perl::Tidy
 
       - name: Run Tests
         env:
           AUTHOR_TESTING: 1
           RELEASE_TESTING: 1
-        run: auto-build-and-test-dist
+        run: perl Makefile.PL && make test
```

Comment:
The `auto-build-and-test-dist` only comes for free with the Docker images.

Reply:
Since you're only using Ubuntu here you could use the Docker images for all of the builds, but not using them does make it easier to catch if you have defined all of your deps correctly etc.